classes/Search.php

```diff
@@ -137,7 +137,23 @@ class SearchCore
 
     public static function extractKeyWords($string, $id_lang, $indexation = false, $iso_code = false)
     {
-        $sanitizedString = Search::sanitize($string, $id_lang, $indexation, $iso_code, false);
+/*         start improvement
+         it allowss the '&' to be indexed for product names like J&B*/
+        if (!stripos($string, '&')) {
+            $sanitizedString = Search::sanitize($string, $id_lang, false, Context::getContext()->language->iso_code);
+        } else {
+            $preWords = explode('&', $string);
```

Comment:
Maybe we could put all of this logic into a function `sanitizeButPreserveAmpersand()`?
```
$sanitizedString .= sanitizeButPreserveAmpersand($string, $id_lang, $indexation = false, $iso_code = false);
```

Reply:
This is a good idea especially if we foresee some other extra signs that would be useful to include in the indexation in the future.
The one I bumped into most frequently was the * because there are a lot of products having it in the name (whisky, shampoo etc ). 
There may be other.